commit cb618d9f9d3c7e16c8575d2b973e964fdfd27c9c (refs/changes/51/3351/3)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2018-02-13T22:29:39+00:00 (1 year, 8 months ago)
    
    TRITON-129 workflow job task throws uncaught exception in some error cases when provisioning volumes
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Approved by: Josh Wilsdon <jwilsdon@joyent.com>

diff --git a/lib/workflows/nfs-volumes/index.js b/lib/workflows/nfs-volumes/index.js
index 962784b..4bee9ae 100644
--- a/lib/workflows/nfs-volumes/index.js
+++ b/lib/workflows/nfs-volumes/index.js
@@ -17,9 +17,7 @@
  */
 
 var vasync = require('vasync');
-var VError = require('verror');
-
-var common = require('../job-common');
+var verror = require('verror');
 
 function provisionNfsVolumes(job, cb) {
     var requiredVolumes;
@@ -77,7 +75,7 @@ function provisionNfsVolumes(job, cb) {
             job.createdVolumes = createdVolumes;
             cb(null, 'All volumes provisioned');
         } else {
-            cb(new VError(err, 'Could not provision volumes'));
+            cb(new verror(err, 'Could not provision volumes'));
         }
     });
 
@@ -337,8 +335,6 @@ function provisionNfsVolumes(job, cb) {
                         next(volumeCreationErr);
                         return;
                     }
-
-                    next(volumeCreationErr);
                 }
             }]}, function onVolumeProvDone(volProvErr) {
                 if (provisionContext.volumeReservation === true &&
